Panic during bootstrap of `controller-manager` due to second call to `controllerruntime.SetupSignalHandler()`

`main` branch currently has a bug in `controller-manager` where two consecutive calls to `controllerruntime.SetupSignalHandler()` exists. This causes a panic during bootstrap.

```
k logs -n ironcore-net-system deployment/ironcore-net-controller-manager
2024-03-11T11:40:52Z    INFO    setup   starting manager
panic: close of closed channel

goroutine 1 [running]:
sigs.k8s.io/controller-runtime/pkg/manager/signals.SetupSignalHandler()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/manager/signals/signal.go:31 +0x1e
main.main()
        /workspace/cmd/controller-manager/main.go:185 +0x17d
```

First: https://github.com/ironcore-dev/ironcore-net/blob/main/cmd/controller-manager/main.go#L66
Second: https://github.com/ironcore-dev/ironcore-net/blob/main/cmd/controller-manager/main.go#L185

--- a/cmd/controller-manager/main.go
+++ b/cmd/controller-manager/main.go
@@ -182,7 +182,7 @@ func main() {
 	}
 
 	setupLog.Info("starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+	if err := mgr.Start(ctx); err != nil {
 		setupLog.Error(err, "problem running manager")
 		os.Exit(1)
 	}

